src/cpprealm/persisted.hpp

```diff
@@ -345,8 +615,13 @@ persisted_base<T>::~persisted_base()
 {
     if constexpr (realm::type_info::property_type<T>() == PropertyType::String) {
         using std::string;
-        if (!m_obj)
-            unmanaged.~string();
+        if (!m_obj) {
+            if constexpr(std::is_same_v<T, std::vector<std::string>>) {
```

Comment:
I'm not sure this is where the constexpr check should be. You should be seeing if the property_type is _any_ kind of  Array and then yes, call the destructor for it.